actors/common/src/main/java/org/sunbird/metrics/actors/CourseMetricsActor.java

```diff
@@ -213,216 +224,206 @@ private void courseProgressMetrics(Request actorMessage) {
-          map.remove(JsonKey.DATE_TIME);
-          map.remove(JsonKey.COURSE_ENROLL_DATE);
-          map.remove(JsonKey.USER_ID);
-          map.remove(JsonKey.BATCH_ID);
+        } else {
+          map.put(JsonKey.USERNAME, null);
+          map.put("org", null);
+          map.put("batchEndsOn", null);
         }
+        map.remove(JsonKey.DATE_TIME);
+        map.remove(JsonKey.COURSE_ENROLL_DATE);
+        map.remove(JsonKey.USER_ID);
+        map.remove(JsonKey.BATCH_ID);
+      }
 
-        Map<String, Object> responseMap = new LinkedHashMap<>();
-        Map<String, Object> userdataMap = new LinkedHashMap<>();
-        Map<String, Object> courseprogressdataMap = new LinkedHashMap<>();
-        Map<String, Object> valueMap = new LinkedHashMap<>();
+      Map<String, Object> responseMap = new LinkedHashMap<>();
+      Map<String, Object> userdataMap = new LinkedHashMap<>();
+      Map<String, Object> courseprogressdataMap = new LinkedHashMap<>();
+      Map<String, Object> valueMap = new LinkedHashMap<>();
 
-        userdataMap.put(JsonKey.NAME, "List of users enrolled for the course");
-        userdataMap.put("split", "content.sum(time_spent)");
-        userdataMap.put("buckets", useresContent);
+      userdataMap.put(JsonKey.NAME, "List of users enrolled for the course");
+      userdataMap.put("split", "content.sum(time_spent)");
+      userdataMap.put("buckets", useresContent);
 
-        courseprogressdataMap.put(JsonKey.NAME, "List of users enrolled for the course");
-        courseprogressdataMap.put("split", "content.sum(time_spent)");
-        courseprogressdataMap.put("buckets", esContent);
+      courseprogressdataMap.put(JsonKey.NAME, "List of users enrolled for the course");
+      courseprogressdataMap.put("split", "content.sum(time_spent)");
+      courseprogressdataMap.put("buckets", esContent);
 
-        valueMap.put("course.progress.users_enrolled.count", userdataMap);
-        valueMap.put("course.progress.course_progress_per_user.count", courseprogressdataMap);
+      valueMap.put("course.progress.users_enrolled.count", userdataMap);
+      valueMap.put("course.progress.course_progress_per_user.count", courseprogressdataMap);
 
-        responseMap.put("period", periodStr);
-        responseMap.put("series", valueMap);
+      responseMap.put("period", periodStr);
+      responseMap.put("series", valueMap);
 
-        Response response = new Response();
-        response.putAll(responseMap);
-        cache.setData(JsonKey.CourseProgress, batchId, periodStr, responseMap);
-        sender().tell(response, self());
-        return;
-      } else {
+      Response response = new Response();
+      response.putAll(responseMap);
+      sender().tell(response, self());
+      return;
+    } else {
 
-        ProjectLogger.log(
-            "CourseMetricsActor-courseProgressMetrics-- Courses for batch is empty .");
+      ProjectLogger.log(
+          "CourseMetricsActor:courseProgressMetrics courses for batch is empty.",
```

Comment:
From:
"CourseMetricsActor:courseProgressMetrics courses for batch is empty."

To:
"CourseMetricsActor:courseProgressMetrics: Course not found for given batchId."